shanoir-ng-datasets/src/main/java/org/shanoir/ng/solr/repository/SolrRepositoryImpl.java

```diff
@@ -0,0 +1,122 @@
+/**
+ * 
+ */
+package org.shanoir.ng.solr.repository;
+
+import java.util.Collection;
+
+import javax.annotation.Resource;
+
+import org.shanoir.ng.shared.dateTime.DateTimeUtils;
+import org.shanoir.ng.solr.model.ShanoirSolrDocument;
+import org.shanoir.ng.solr.model.ShanoirSolrFacet;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.solr.core.SolrTemplate;
+import org.springframework.data.solr.core.query.Criteria;
+import org.springframework.data.solr.core.query.FacetOptions;
+import org.springframework.data.solr.core.query.FacetQuery;
+import org.springframework.data.solr.core.query.Node;
+import org.springframework.data.solr.core.query.SimpleFacetQuery;
+import org.springframework.data.solr.core.query.result.FacetPage;
+import org.springframework.data.solr.core.query.result.SolrResultPage;
+
+/**
+ * @author yyao
+ *
+ */
+public class SolrRepositoryImpl implements SolrRepositoryCustom{
+	@Resource
+	private SolrTemplate solrTemplate;
+	
+	@Override
+	public SolrResultPage<ShanoirSolrDocument> findByFacetCriteria(ShanoirSolrFacet facet, Pageable pageable) {
+		Criteria criteria = new Criteria(Criteria.WILDCARD).expression(Criteria.WILDCARD);
+		return getSearchResultsWithFacets(criteria, facet, pageable);
+	}
+	
+	@Override
+	public SolrResultPage<ShanoirSolrDocument> findByStudyIdInAndFacetCriteria(Collection<Long> studyIds, 
+			ShanoirSolrFacet facet, Pageable pageable) {
+		Criteria criteria = new Criteria("studyId").in(studyIds);
+		return getSearchResultsWithFacets(criteria, facet, pageable);
+		
+	}
+	
+	private SolrResultPage<ShanoirSolrDocument> getSearchResultsWithFacets(Criteria criteria, ShanoirSolrFacet facet, Pageable pageable) {
+		
+		if (facet.getStudyName() != null && facet.getStudyName().size() > 0) {
```

Comment:
use isEmpty()

Reply:
for this list, is not null and isEmpty is not the same, right? only use isEmpty() could get nullpointerException when the list is null

Reply:
Yep, it was just a little change from size() > 0 to isEmpty()

Reply:
ah ok! I,ll do it right now